### ISSUE TITLE
Validation error: decodeUTF16String: not UTF16BE

### ISSUE THREAD

**Issue by Devessier** [pdfcpu/pdfcpu#264]
Hello everybody!
I generate PDF files using [gofpdf](https://github.com/phpdave11/gofpdf) and merge some of them using this great library.
Most of the time everything works well, my PDF files are merged correctly.
But for one specific file, pdfcpu throws an error and the merge operation does not succeed.
This file correctly opens in a browser or in Preview macOS app.

I ran the command `pdfcpu validate -vv` and the output is the following:
[crash.log](https://github.com/pdfcpu/pdfcpu/files/5690137/crash.log)

I'm using the latest version of pdfcpu.
I can send the concerned PDF file by email.

I thank you for your great job.
Best regards
Baptiste Devessier

**Comment by hhrutter** [comment]
Hi, thanks for reporting this.
Looks like a bug.
Yes, please send me a file you can share and I provide a fix.

**Comment by Devessier** [comment]
Thank you, I sent you the document.
The document I'm talking about has been electronically signed.
Without the electronic signature no error is returned by pdfcpu.

**Comment by hhrutter** [comment]
This is iText messing things up when patching the info dict during whatever iText operation is applied - signing a document in this case obviously. iText wants to modify the *Producer* field of the info dict which is of type *text string*.

The PDF specification defines a *text string* to be either a PDFDocEncoded byte based string or an UTF-16BE string introduced by the 2 byte BOM #FEFF: 
<img width="661" alt="Screenshot 2020-12-16 at 01 41 20" src="https://user-images.githubusercontent.com/11322155/102289841-e86ed800-3f3f-11eb-8f75-8fb7591c232f.png">


The short explanation goes like this:

The original hexadecimal content of the key *Producer* in the info dict:
```
28FEFF00 46005000 44004600 20003100 2E003729
```
This equals "FPDF 1.7" - a correct UTF-16BE byte sequence (omitting the bytes for the enclosing brackets of string literals)

The patched content ends up being:
```
28FEFF46 50444620 312E373B 206D6F64 69666965 64207573 696E6720 69546578 7420322E 312E3720 62792031 54335854 29
```
which equals "{BOM}FPDF 1.7; modified using iText 2.1.7 by 1T3XT"

In the defense of iText we can tell it gracefully wants to append itself to the content of *Producer* but unfortunately it does not recognize or ignores the fact that the original *text string* was written in correct UTF-16BE using the #FEFF BOM and 2 bytes for each character.

There are two ways for handling this in a way that conforms to the PDF spec:
* Rewrite the modified string in UTF-16BE with BOM
* Rewrite the modified string PDFDocEncoded and byte based (without BOM!)

Since the original PDFWriter was using UTF-16 there probably must have been a reason for it so 
IMHO the result of this modification should be an UTF-16 as well.

Still iText chose to rewrite this field using PDFDocEncoding and for some reason carried over the existing BOM which is undefined in PDFDocEncoding.

The result is invalid since any PDF Reader recognizing a BOM in a string literal expects an even number of bytes afterwards since UTF-16 uses 2 bytes for each character.

Since we can't work on the assumption that we are dealing with a PDFDocEncoded string with 2 invalid leading bytes #FEFF only because we failed to interprete a UTF-16BE byte sequence because of an odd number of bytes (instead of even) we are not in a position to repair this string and can only ignore this error during validation. (If there was an even number of bytes this issue would not even show up - weirdly so!)

Fix goes into the next release which should be out by Xmas.

Interesting stuff!

Thanks for using pdfcpu 💚

**Comment by hhrutter** [comment]
Since this is going to be an ugly hack, another suggestion:

Is there a way for you to switch to a more recent version of iText for signing or to another tool?

iText, v2.1.7 - the one you are using - is known to have issues and more than 10 years old. 
There is a good chance this issue has been fixed in iText by now.

**Comment by Devessier** [comment]
I thank you for your long explanation and your time!
I can not really change how electronic signature is achieved because I use a SaaS to sign my documents.

I tried to merge the same documents using other tools than pdfcpu.
I tried to do it with [pdfunite from poppler](https://manpages.debian.org/testing/poppler-utils/pdfunite.1.en.html) and [iLovePDF website](https://www.ilovepdf.com) and for both the documents could be merged.
I don't know how they solved the problem you point but I think it would be important for pdfcpu to handle this case.
